bin/test-example

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+# need bash, not sh, for default variable ${var:-default}
+
+exampledir=$1
+
+examplename=$(basename "$exampledir")
+exercisedir=$(dirname $(dirname "$exampledir"))
+exercisename=$(basename "$exercisedir")
+
+buildfolder="${TRAVIS_BUILD_DIR:-.}/build/${exercisename}/${examplename}"
+mkdir -p "${buildfolder}"
+cp -rL ${exercisedir}/stack.yaml ${exercisedir}/test ${exampledir}/* "${buildfolder}"
+
+cd $buildfolder
+
+examplecache="${HOME}/.foldercache/${exercisename}/${examplename}/.stack-work"
```

Comment:
OK to create a .foldercache in user's personal HOME if running locally? If not, how do we deal with this? `if [ -n "$TRAVIS" ] examplecache="something-else"` perhaps?